pkg/controller/jobframework/integrationmanager.go

```diff
@@ -265,7 +267,10 @@ func GetIntegration(name string) (IntegrationCallbacks, bool) {
 func GetIntegrationByGVK(gvk schema.GroupVersionKind) (IntegrationCallbacks, bool) {
 	for _, name := range manager.getList() {
 		integration, ok := GetIntegration(name)
-		if ok && gvk == integration.NewJob().GVK() {
+		if ok && integration.NewJob == nil && integration.GVK == gvk {
+			return integration, true
+		}
+		if ok && integration.NewJob != nil && integration.NewJob().GVK() == gvk {
 			return integration, true
 		}
```

Comment:
I would like to use @mimowo [suggestion](https://github.com/kubernetes-sigs/kueue/pull/2988#discussion_r1745789652).

```
func matchingGVK(integration, gvk) bool {
  if integration.NewJob != nil {
      return gvk == integration.NewJob().GVK()
   } else {
    return gvk == integration.GVK
  }
}
```

WDYT?

Reply:
yes, missed that, updated now.